### PR TITLE
Various stuff for upcoming release

### PR DESCRIPTION
**Pull request by franz**
Updated documentation, fixed a few bugs, etc.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -8,6 +8,25 @@ Highlights
   well as the number of compute units available on the device.
 - Support for NVIDIA GPUs via a new CUDA backend (currently experimental).
 - Removed support for BBVectorizer
+- LLVM 5.0 support
```
Comment (pjaaskel) [inline CHANGES]: Can you make the items full sentences with a capital first letter? Also, did you scan through the log for changes? The version shall be 1.0.

File: README.packaging [hunk 2]
```diff
@@ -6,11 +6,18 @@ Pocl should probably be built with ICD enabled for desktop
...
 on one would be beneficial.
 
-The pocl.icd file (which the ICD loader uses to load the pocl lib)
-has a full path to the installed libpocl.so file. When building
-for a multiarch, remove the "@libdir@/" from pocl.icd.in before
-building. This way the system library loader can pick up the correct
-architecture library.
+CMake options for a distribution build
+--------------------------------------
+- ``-DKERNELLIB_HOST_CPU_VARIANTS=distro``
```
Comment (pjaaskel) [inline README.packaging]: I think this is a useful feature to include in the change list.

File: doc/sphinx/source/conformance.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+- the "not" operator test (``math_brute_force/bruteforce not``) may fail to
+  compile with LLVM 4.0 with certain vector sizes on some hardware.
+  This does not seem to affect the rest of the testsuite in any way, and
+  appears to be fixed with LLVM 5.0
+
+- a few tests from ``basic/test_basic`` may fail / segfault because they
+  request a huge amount of memory for buffers.
+
+.. _sigfpe-handler:
```
Comment (pjaaskel) [inline doc/sphinx/source/conformance.rst]: I think this anchor is in a wrong place, should be just before the issue item?

File: doc/sphinx/source/conformance.rst [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  options are honored.
+
+- Many of ``native_`` and ``half_`` variants of kernel library functions are mapped
+  to the "full" variants.
+
+- the optional OpenGL interop extension is not supported
+
+Conformance tests results (kernel library precision) on tested hardware
+-----------------------------------------------------------------------
+
```
Comment (pjaaskel) [inline doc/sphinx/source/conformance.rst]: Add a date of executing the test and the LLVM version here for reference.

File: doc/sphinx/source/conformance.rst [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+            ceil            0.00                      0x0p+0
+            ceilD           0.00                      0x0p+0
+        copysign            0.00                      {0x0p+0, 0x0p+0}
+        copysignD           0.00                      {0x0p+0, 0x0p+0}
+             cos            2.37                      0x1.1338ccp+20
+             cosD           2.27                      -0x1.d10000000074p+380
+            cosh            2.41                      -0x1.602166p+2
+            coshD           1.43                      -0x1.98000000003efp+5
+           cospi            1.94                      0x1.d73b56p-2
+           cospiD           2.46                      -0x1.adffffffffa91p-2
```
Comment (pjaaskel) [inline doc/sphinx/source/conformance.rst]: Are these errors still within the allowed limits?

File: examples/conformance/CMakeLists.txt [hunk 6]
```diff
@@ -883,6 +1172,201 @@ add_test(NAME "${TS_NAME}_conversion_fast_double_ulong"
 add_test(NAME "${TS_NAME}_conversion_fast_long_double"
          COMMAND "${TS_BUILDDIR}/test_conformance/conversions/test_conversions" -m -4 long_double long_sat_double long_rte_double long_sat_rte_double long_rtp_double long_sat_rtp_double long_rtn_double long_sat_rtn_double long_rtz_double long_sat_rtz_double )
 
+#***************************************************************************
+
+# tests math on vectors of 4 floats.
+# This is usually the fastest version on machines with AVX(2)
+
+#add_test(NAME "${TS_NAME}_math_fast_add"
```
Comment (pjaaskel) [inline examples/conformance/CMakeLists.txt]: What are the commented out tests?

File: lib/CL/devices/basic/basic.c [hunk 7]
```diff
@@ -208,9 +215,13 @@ pocl_basic_init_device_infos(unsigned j, struct _cl_device_id* dev)
 #ifdef _CL_DISABLE_DOUBLE
   dev->double_fp_config = 0;
 #else
+  /* TODO: all of these are the minimum mandated, but not all CPUs may actually
+   * support all of them. */
   dev->double_fp_config = CL_FP_FMA | CL_FP_ROUND_TO_NEAREST
                           | CL_FP_ROUND_TO_ZERO | CL_FP_ROUND_TO_INF
                           | CL_FP_INF_NAN | CL_FP_DENORM;
+  /* this is a workaround for a bug in https://github.com/Oblomov/clinfo */
```
Comment (pjaaskel) [inline lib/CL/devices/basic/basic.c]: Can you expand a bit: which bug and why it fixes it?

File: lib/CL/pocl_llvm_utils.cc [hunk 8]
```diff
@@ -152,14 +153,43 @@ char *get_cpu_name() {
...
   bool res = llvm::sys::getHostCPUFeatures(features);
   assert(res);
   return ((features["fma"] || features["fma4"]) ? 1 : 0);
 }
-#endif
+
+#define VECWIDTH(x)                                                            \
+  std::min(std::max((lane_width / (unsigned)(sizeof(x))), 1U), 16U)
+
+void cpu_setup_vector_widths(cl_device_id dev) {
```
Comment (pjaaskel) [inline lib/CL/pocl_llvm_utils.cc]: I think this belongs to the same place directory where the hwloc code is. You use LLVM's infra for detecting vector types for basic/pthread only as it works only with the host CPU, right?

File: doc/sphinx/source/conformance.rst (hunk 3 above)
Comment (franz) [reply doc/sphinx/source/conformance.rst]: it should, and was there originally, but sphinx screams and refuses to make html :[

File: examples/conformance/CMakeLists.txt (hunk 6 above)
Comment (franz) [reply examples/conformance/CMakeLists.txt]: ah, forgotten. i'll uncomment them

File: doc/sphinx/source/conformance.rst (hunk 5 above)
Comment (franz) [reply doc/sphinx/source/conformance.rst]: yes, cosh & cospi are <= 4 ULP. this is formatted output from the math conformance test

File: lib/CL/pocl_llvm_utils.cc (hunk 8 above)
Comment (franz) [reply lib/CL/pocl_llvm_utils.cc]: > You use LLVM's infra for detecting vector types for basic/pthread only 

yes. The difference is, hwloc code is independent of LLVM (and works on LLVM-less builds) - this code is not. If i moved this to hwloc, i'd have to either make it depend on LLVM there (i'd rather not), or copy some LLVM code that detects the cpu features. If you want this in hwloc code, i strongly prefer the latter (copy).

File: README.packaging (hunk 2 above)
Comment (franz) [reply README.packaging]: well, it's been in previous releases already, but i see it's not documented. Will fix

File: examples/conformance/CMakeLists.txt (hunk 6 above)
Comment (pjaaskel) [reply examples/conformance/CMakeLists.txt]: Do you mean they are not required by the conformance or are already included in some other suite?

File: lib/CL/pocl_llvm_utils.cc (hunk 8 above)
Comment (pjaaskel) [reply lib/CL/pocl_llvm_utils.cc]: Fair enough.

**Pull request merged**